gearcatalog/frontend/src/components/common/Input.tsx

```diff
@@ -1,10 +1,10 @@
-import React from "react";
+import React, { SetStateAction } from "react";
 
 type InputProps = {
   name: string,
   val: string | number,
   type: string,
-  setVal: (val: string) => void,
+  setVal: (val: SetStateAction<string> | SetStateAction<number>) => void,
```

Comment:
Check out generics for this. I believe this struct could be reshaped like this:

```ts
type InputProps<ValueType> = {
  name: string,
  val: ValueType,
  setVal: (val: ValueType) => void,
  ...
};
```

Then the input class would also be generisized in the same way.

```ts
const Input = <ValueType,>(props: InputProps<ValueType>): JSX.Element => {
  ...
};
```

Note that it is `<ValueType,>`, with a trailing comma, for the function. This is to break out of JSX's element parsing.

The bonus in doing it this way is it ensures correct typing between the `val` and `setVal` properties.